components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -77,7 +78,7 @@ abstract class AbstractFetchDownloadService : Service() {
     @VisibleForTesting
     internal val context: Context get() = this
 
-    internal var downloadJobs = mutableMapOf<Long, DownloadJobState>()
+    internal var downloadJobs = ConcurrentHashMap<Long, DownloadJobState>()
```

Comment:
This map isn't actually accessed from different threads, or am I missing something in your change. Even the new https://github.com/mozilla-mobile/android-components/pull/6458/files#diff-03002a5338a9935256791a2b2770b720R241 launches it on the `MainScope` so that's still the main thread, no?